### PR TITLE
Update tests to remove temporary data upon completion.

### PR DESCRIPTION
**Pull request by gbggrant**

### CODE REVIEW

File: src/test/java/picard/vcf/filter/TestFilterVcf.java [hunk 1]
```diff
@@ -180,6 +178,23 @@ File testFiltering(final File vcf, final String outputExtension, final double mi
         return out;
     }
 
+    private File createTempVcfFileAndDeleteItAndItsIndexFileOnExit(final String prefix, final String suffix) throws IOException {
```
Comment (yfarjoun) [inline src/test/java/picard/vcf/filter/TestFilterVcf.java]: that's quite a name..I think that temporary might be enough indication that the file will be deleted. I'd go with createTemporaryIndexedVcfFile() and you can javadoc it to explain.

File: src/test/java/picard/vcf/FixVcfHeaderTest.java [hunk 2]
```diff
@@ -27,31 +27,47 @@
...
+        OUTPUT_VCF                   = new File(TEST_DATA_PATH, "output.vcf");
+        HEADER_VCF                   = new File(TEST_DATA_PATH, "header.vcf");
+        HEADER_VCF_WITH_EXTRA_SAMPLE = new File(TEST_DATA_PATH, "header_with_extra_sample.vcf");
+    }
 
     private void runFixVcfHeader(final int checkFirstNRecords,
                                  final File replacementHeader,
                                  final boolean enforceSampleSamples) {
         final FixVcfHeader program = new FixVcfHeader();
         final File outputVcf = new File(OUTPUT_DATA_PATH, "output.vcf");
```
Comment (yfarjoun) [inline src/test/java/picard/vcf/FixVcfHeaderTest.java]: why not use createTempVcfFIleAndDeleteItAndItsIndex here as well?

Comment (gbggrant) [inline src/test/java/picard/vcf/FixVcfHeaderTest.java]: That's a good idea.

**Comment by gbggrant** [comment]
Hi @yfarjoun I have responded to your comments.  Please take a look when you have a chance.

**Comment by gbggrant** [comment]
Hi @yfarjoun Can you take another look?

File: src/test/java/picard/vcf/VcfTestUtils.java [hunk 3]
```diff
@@ -12,6 +12,31 @@
...
 public class VcfTestUtils {
+
+    /**
+     * This method creates a temporary VCF file and it's appropriately named index file, and will delete them on exit.
+     * @param prefix - The prefix string to be used in generating the file's name; must be at least three characters long
+     * @param suffix - The suffix string to be used in generating the file's name; may be null, in which case the suffix ".tmp" will be used
+     * @return A File object referencing the newly created temporary VCF file
+     * @throws IOException - if a file could not be created.
+     */
+    public static File createTemporaryIndexedVcfFile(final String prefix, final String suffix) throws IOException {
```
Comment (yfarjoun) [inline src/test/java/picard/vcf/VcfTestUtils.java]: the two functions createTemporaryIndexedVcfFile and createIndexedVcf do different things, but have misleadingly similar names...could you make sure that the names make it more clear what the differences are, or alternatively, rebase them into one function with an additional argument?

**Comment by yfarjoun** [comment]
@gbggrant what the status here?

**Comment by yfarjoun** [comment]
bump

**Comment by yfarjoun** [comment]
@gbggrant ???

**Comment by vdauwera** [comment]
_crickets_

**Comment by gbggrant** [comment]
@vdauwera Thanks for the wake-up crickets.
@yfarjoun I made a small change to rename the method as you suggested and used it elsewhere.  Do you want to do a last check?

**Comment by yfarjoun** [comment]
😲 💯 best PR ever!!! AAA+++ will review again.

**Pull request merged**